test-framework/mongodb/src/main/java/io/quarkus/test/mongodb/MongoDatabaseTestResource.java

```diff
@@ -0,0 +1,89 @@
+package io.quarkus.test.mongodb;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.IMongodConfig;
+import de.flapdoodle.embed.mongo.config.MongoCmdOptionsBuilder;
+import de.flapdoodle.embed.mongo.config.MongodConfigBuilder;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.process.runtime.Network;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+/**
+ * Resource to provide a mongo database instance during the execution of the Tests.
+ * 
+ * @see QuarkusTestResource
+ */
+public class MongoDatabaseTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private static final Logger LOGGER = Logger.getLogger(MongoDatabaseTestResource.class);
+
+    private static final String CONFIGURED_CONNECTION_STRING = "mongodb.connection_string";
```

Comment:
Do we have this pattern in another one?

Reply:
No, the only similar one that I found was in MongoTestBase.java in mongodb-client extension tests, and it was using `connection_string`.
I qualified it to make it more clear when you pass the System property.